### PR TITLE
Fixed const correctness of gpioInit functions.

### PR DESCRIPTION
**Pull request by martinbudden**
Addresses issue https://github.com/cleanflight/cleanflight/issues/1797

### CODE REVIEW

File: src/main/drivers/gpio.h [hunk 1]
```diff
@@ -110,9 +110,9 @@ typedef struct
 static inline void digitalHi(GPIO_TypeDef *p, uint16_t i) { p->BSRR = i; }
 static inline void digitalLo(GPIO_TypeDef *p, uint16_t i)     { p->BRR = i; }
 static inline void digitalToggle(GPIO_TypeDef *p, uint16_t i) { p->ODR ^= i; }
-static inline uint16_t digitalIn(GPIO_TypeDef *p, uint16_t i) {return p->IDR & i; }
+static inline uint16_t digitalIn(const GPIO_TypeDef *p, uint16_t i) {return p->IDR & i; }
```
Comment (ledvinap) [inline src/main/drivers/gpio.h]: The *p is effectively `const volatile`, which is a bit strange ..

Comment (hydra) [inline src/main/drivers/gpio.h]: yeah, this change doesn't fit with the rest of the digital* functions.  I agree that the gpio_config changes should be marked as const though.

Comment (ledvinap) [inline src/main/drivers/gpio.h]: Yes, gpio_config_t must be `const` (can be placed in ROM)

**Comment by martinbudden** [comment]
@hydra 
I don't understand why you say the `digitalIn` change doesn't fit in. The `const` here just says that the `digitalIn` function does not change anything that `p` points to. Which it doesn't. Now presumably the thing `p` points to is `volatile`, which as @ledvinap says is a bit strange. But surely the compiler is sensible enough not to optimise away this volatility.

Sure, it makes the signature of the `digitalIn` function different from the other `digital*` functions, but that's the point - the other functions are output functions and deliberately change the thing `p` points to.

I'd like to understand your objection.

**Comment by martinbudden** [comment]
I've now removed the `const` specifier on the `digitalIn` function.

**Pull request merged**